### PR TITLE
dm-verity hash output

### PR DESCRIPTION
**Pull request by gtank**
Branch adds option to disk_util for dumping verity hashes in the build directory

### CODE REVIEW

File: build_library/prod_image_util.sh [hunk 1]
```diff
@@ -87,7 +87,8 @@ EOF
   # Make the filesystem un-mountable as read-write and setup verity.
   if [[ ${disable_read_write} -eq ${FLAGS_TRUE} ]]; then
     "${BUILD_LIBRARY_DIR}/disk_util" --disk_layout="${disk_layout}" \
-      verity "${BUILD_DIR}/${image_name}"
+      verity "${BUILD_DIR}/${image_name}" | \
+      tee "${BUILD_DIR}/verity_info"
```
Comment (marineam) [inline build_library/prod_image_util.sh]: How about add support to disk_util for parsing the output of veritysetup and add an option to specify where to write the hash to. Something like:

```sh
disk_util --disk_layout="${disk_layout}" verity \
  --root_hash="${BUILD_DIR}/${image_name%.bin}_verity.txt" \
  "${BUILD_DIR}/${image_name}"
```

would create a file named `coreos_production_image_verity.txt` containing the simple hash:

```
bd4cf58d66bb3ffde9bcb65bbd6b82303034b181106aa83f521864a0c3cddbad
```

If you put those changes in a PR that doesn't include the experimental grub bit we can merge it any time. The grub changes will need to wait until the hash is in the kernel.

File: build_library/disk_util [hunk 2]
```diff
@@ -351,6 +351,22 @@ def Sudo(cmd, stdout_null=False):
     if null:
       null.close()
 
+def SudoPipe(cmd):
```
Comment (marineam) [inline build_library/disk_util]: Since it is returning a string instead of a pipe perhaps name this SudoOutput instead.

Comment (gtank) [inline build_library/disk_util]: sounds good to me

File: build_library/disk_util [hunk 3]
```diff
@@ -739,12 +755,23 @@ def Verity(options):
...
-	    '--hash-offset', part['fs_bytes'],
-	    loop_dev, loop_dev])
+      verityout = SudoPipe(['veritysetup', 'format', '--hash=sha256',
+                  '--data-block-size', part['fs_block_size'],
+                  '--hash-block-size', part['fs_block_size'],
+                  '--data-blocks', part['fs_blocks'],
+                  '--hash-offset', part['fs_bytes'],
+                  loop_dev, loop_dev])
+      root_line = verityout.strip().split("\n")[-1]
+      root_hash = root_line.split("\t")[-1]
```
Comment (marineam) [inline build_library/disk_util]: Although this is pretty simple I'd like to do a little more verification of the output. perhaps:

```python
print verityout.strip()
m = re.search("Root hash:\s+([a-f0-9]{64})$", verityout, re.IGNORECASE|re.MULTILINE)
if not m:
    raise Exception("Failed to parse verity output!")
if options.root_hash:
    with open(options.root_hash, "w") as hash_file:
        hash_file.write(m.group(1))
        hash_file.write("\n")
```

**Comment by marineam** [comment]
LGTM, please squash your commits before merging since there is older unrelated GRUB stuff in there still.

**Pull request merged**